repos/system_upgrade/el7toel8/actors/unsupportedupgradecheck/actor.py

```diff
@@ -0,0 +1,74 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class UnsupportedUpgradeCheck(Actor):
+    """
+    Checks enviroment variables and produces a warning report if the upgrade is unsupported.
+
+    Upgrade is unsupported if any LEAPP_DEVEL_* variable is used or an experimental actor is enabled.
+    This can be overridden by setting the variable LEAPP_UNSUPPORTED (at user's own risk).
+    """
+
+    name = 'unsupported_upgrade_check'
+    consumes = ()
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        leapp_vars = self.configuration.leapp_env_vars
+        devel_vars = [v for v in leapp_vars if v.name.startswith('LEAPP_DEVEL_')]
+        experimental = bool([v for v in leapp_vars if v.name == 'LEAPP_EXPERIMENTAL' and v.value == '1'])
+        override = bool([v for v in leapp_vars if v.name == 'LEAPP_UNSUPPORTED' and v.value == '1'])
+
+        if override:
+            reporting.create_report([
+                reporting.Title('Upgrade is unsupported'),
+                reporting.Summary(
+                    'Environment variable LEAPP_UNSUPPORTED has been detected. A successful and safe '
```

Comment:
Love the change, thanks! Booting a vm for a fullstack upgrade run, should get the results tomorrow the latest.

Reply:
Sorry for taking too long to confirm, works as expected